src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncMainTests.cs

```diff
@@ -1479,5 +1479,439 @@ static class Program {
                 //     static Task Main() {
                 Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "Task").WithArguments("MyExtensions.GetAwaiter(System.Threading.Tasks.Task)", "test").WithLocation(31, 12));
         }
+
+        [Fact]
+        [WorkItem(33542, "https://github.com/dotnet/roslyn/issues/33542")]
+        public void AwaitInFinallyInNestedTry_01()
+        {
+            string source =
+@"using System.Threading.Tasks;
+class Program
+{
+    static async Task Main()
+    {
+        try
+        {
+            try
+            {
+                return;
+            }
+            finally
+            {
+                await Task.CompletedTask;
+            }
+        }
+        catch
+        {
+        }
+        finally
+        {
+            await Task.CompletedTask;
+        }
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe.WithOptimizationLevel(OptimizationLevel.Release));
+            var verifier = CompileAndVerify(comp, expectedOutput: "");
```

Comment:
>expectedOutput: "" [](start = 50, length = 18)

Consider having adjusting the test scenario so that by looking at the output we could confirm that the behavior at runtime is what we expect. It is really hard to confirm that from the IL and will be even harder to confirm if IL would change in the future in some insignificant way.

Reply:
I'd rather keep the test simple to ensure we hit the failing scenario.

---
In reply to: [272674792](https://github.com/dotnet/roslyn/pull/34791#discussion_r272674792) [](ancestors = 272674792)